crates/bevy_animation/src/_props.rs

```diff
@@ -0,0 +1,56 @@
+// TODO: Implement safe property path like `Transforms::props().translation().x()` using only const, (Property validation is working at least)
+
+/// A type safe way of referring to properties that can be animated.
+///
+/// If a type implements `AnimatedProperties` each property can be retrieved
+/// by `Transform::props().translation().x()`
+pub struct Prop<T> {
+    pub name: Cow<'static, str>,
+    _marker: PhantomData<T>,
+}
+
+impl<T> Prop<T> {
+    pub const fn borrowed(name: &'static str) -> Self {
+        Prop {
+            name: Cow::Borrowed(name),
+            _marker: PhantomData,
+        }
+    }
+
+    pub const fn owned(name: String) -> Self {
+        Prop {
+            name: Cow::Owned(name),
+            _marker: PhantomData,
+        }
+    }
+}
+
+// ? NOTE: That upsets me, it should be auto generated path
```

Comment:
Is this the draft to animated nested properties? I feel like this should be possible with the `syn` crate and/or a macro.

Reply:
The function of this `Prop` is to associate each property with a unique type and also avoid misspells; I did tried using proc_macro with no success, thats why I removed from now